automatic/mysql/update.ps1

```diff
@@ -1,48 +1,57 @@
-Import-Module Chocolatey-AU
+﻿Import-Module Chocolatey-AU
 
-$releases = 'https://dev.mysql.com/downloads/mysql/'
+# The mysql-server repo contains a bunch of tags for "mysql-cluster-VERSION"
+# If I use https://api.github.com/repos/mysql/mysql-server/tags then I'll need to paginate to find "mysql-VERSION"
+# I'm using https://api.github.com/repos/mysql/mysql-server//git/matching-refs/tags so that I'll get a complete list without needing to paginate
+# https://stackoverflow.com/a/74561918
+$releases = 'https://api.github.com/repos/mysql/mysql-server/git/matching-refs/tags'
 
 function global:au_SearchReplace {
-    @{
-        'tools\chocolateyInstall.ps1' = @{
-            "(^[$]url\s*=\s*)('.*')"      = "`$1'$($Latest.URL64)'"
-            "(^[$]checksum\s*=\s*)('.*')" = "`$1'$($Latest.Checksum64)'"
-            "(^[$]checksumType\s*=\s*)('.*')" = "`$1'$($Latest.ChecksumType64)'"
-        }
+  @{
+    'tools\chocolateyInstall.ps1' = @{
+      "(^[$]url\s*=\s*)('.*')"          = "`$1'$($Latest.URL64)'"
+      "(^[$]checksum\s*=\s*)('.*')"     = "`$1'$($Latest.Checksum64)'"
+      "(^[$]checksumType\s*=\s*)('.*')" = "`$1'$($Latest.ChecksumType64)'"
     }
- }
+  }
+}
 
 function global:au_GetLatest {
-    $download_page = Invoke-WebRequest -Uri $releases
+  $headers = @{}
+  if (Test-Path Env:\github_api_key) {
+    $headers.Authorization = "token " + $env:github_api_key
+  }
+  $download_page = Invoke-RestMethod -Uri $releases -Headers $header
+
+  $valid_tags = $download_page | Where-Object { $_.ref -notmatch '^refs/tags/mysql-cluster-' }
+  $latest_valid_tag = $valid_tags[-1].ref
+  $latest_valid_version = ($latest_valid_tag -split "/")[-1]
 
-    $version = ($download_page.ParsedHtml.getElementsByTagName('h1') | Where-Object innerhtml -match "^MySQL Community Server ").innerhtml -replace "^MySQL Community Server "
-    $versiondata = Get-Version($version)
-    $version = $versiondata.toString()
+  $versiondata = Get-Version($latest_valid_version)
+  $version = $versiondata.toString()
 
-    $url = 'https://dev.mysql.com/get/Downloads/MySQL-' + $versiondata.toString(2) + '/mysql-' + $version + '-winx64.zip'
-    $Latest = @{ URL64 = $url; Version = $version }
-    return $Latest
+  $url = 'https://cdn.mysql.com/Downloads/MySQL-' + $versiondata.toString(2) + '/mysql-' + $version + '-winx64.msi'
```

Comment:
Switching from ZIP installer to MSI would imply changes to the package's internal work. This is not desirable as this has been and always will be a portable package. 

In the past, the MSI file was there and went away and came back and that 100 times over. To keep this package on track, the Windows ZIP fille should still be used, if available.

Reply:
The ZIP file is available.  Switching to MSI was sloppy on my part, when I copy/pasted from `mysql.workbench`.

2a49231cfd48471e7545da7c2f7edf4e28d93b0d